app/controllers/pets_controller.rb

```diff
@@ -26,8 +31,12 @@ def edit
   def update
     shelter = Shelter.find(params[:shelter_id])
     pet = Pet.find(params[:pet_id])
-    pet.update(pet_params)
-    redirect_to "/shelters/#{shelter.id}/pets/#{pet.id}"
+    if pet.update(pet_params)
```

Comment:
Adds a conditional and flash messages to indicate which fields were missing when submitting the form